W-21215932-Update-Mule-runtime-content-CP (#583)
mulesoft/docs-code-builder: modules/ROOT/pages/int-versions.adoc

//used in procedures::open-project: Open your implementation or integration project//
Set Mule runtime engine (Mule), Java, and connector versions to use in your projects.

> Anypoint Code Builder includes bundled Mule runtime versions that enable you to open and design projects instantly without mandatory runtime downloads. The IDE uses these bundled Mule runtime versions for editing and DataSense. Exact Mule runtime versions are only required when you run or debug your applications.

Reviewer: ```suggestion
Anypoint Code Builder includes bundled Mule runtime versions that enable you to open and design projects instantly without mandatory Mule runtime downloads. The IDE uses these bundled Mule runtime versions for editing and DataSense. Exact Mule runtime versions are only required when you run or debug your applications.
```

Author: OK! Thanks!

* <<bundled-runtimes, Bundled runtimes>>: Learn about the runtimes included with Anypoint Code Builder.

* <<default-versions, Default versions>>: Set default Mule and Java versions to show in version dropdown menus. 

* Project-level versions: Set versions to use in a specific project.
** For Mule and Java versions, see <<project-version-mule-java>>.
** For connector versions, see <<project-versions-connectors>>.

* <<install-additional-runtimes, Install additional runtimes>>: Download specific Mule runtime versions when needed for run or debug operations.  

[[bundled-runtimes]]
== Bundled Mule Runtimes

Anypoint Code Builder includes two bundled Mule runtimes that are available immediately after installation:

* **Mule 4.8 (Java 8 compatible, fallback)**: A minimal runtime bundle compatible with all Java 8-based Mule 4.4 and later projects. This runtime is used for design-time activities with Java 8 projects and serves as the default fallback for legacy applications. You can't use this version to run Mule applications.

> * **Latest Runtime (Java 17)**: The most recent supported Mule runtime version (Edge or LTS) at the time of the Anypoint Code Builder release. This runtime supports Java 17 and can be used for both designing and running applications. Projects configured for Mule 4.6, 4.8, 4.9, and 4.10 are compatible with this bundled runtime.

Reviewer 2: fyi, by the time we release, 4.8 is removed from our supported runtime list

Author: OK! I'll remove 4.8 from our docs. Thanks for letting me know!

Author: @WilliamYLin, what Mule runtime version is bundled in ACB for Java 8 projects?

Reviewer 2: It is a **reduced** version of runtime 4.8, **not a full** runtime. It's essentially the dependencies required for the design service to start, not a standalone runtime

Author: OK! Thanks!

The IDE automatically selects the appropriate bundled Mule runtime version based on your project's Java version:
* Java 8 projects use the bundled 4.8 minimal Mule runtime version
* Java 17 projects use the latest bundled Mule runtime version

These bundled Mule runtime versions guarantee that the design service can always start and serve the canvas and design-time needs without requiring runtime downloads. Projects open instantly using these bundled Mule runtime versions. Download a specific Mule runtime version if you want to run or debug your application with an exact version that isn't available locally.

[[default-versions]]
== Select Default Mule and Java Versions

Select the default Mule and Java versions for new implementation and integration projects. The defaults appear in drop-down menus from which you select a version, for example, when creating a project. 

> The dropdown shows only locally available Mule runtime versions, including the bundled Mule runtime versions. Versions that don't support a feature typically append a message, such as *AsyncAPI implementation unsupported* for Mule 4.4. 

Reviewer: ```suggestion
The dropdown shows only locally available Mule runtime versions, including the bundled ones. Versions that don't support a feature typically append a message, such as *AsyncAPI implementation unsupported* for Mule 4.4. 
```

Author: OK! Thanks!

[[project-version-mule-java]]
== Select Mule and Java Versions for a Project

Set Mule or Java versions for a specific implementation or integration project from *Project Properties*. You can change versions at any time, but test your projects with the versions you intend to deploy. 

> The IDE uses bundled runtimes to open and edit projects immediately. The `minMuleVersion` in your project's `mule-artifact.json` file specifies the minimum compatible Mule runtime version required to open and edit the project. The design service automatically selects any locally available runtime that satisfies this minimum constraint, enabling forward compatibility.

Reviewer 2: The design service uses the **latest** java compatible runtime available in local. Since we bundle the latest runtime (4.11) for java 17 and bundle reduced 4.8 runtime for java 8, this guarantees that we always have a runtime to start design service and serve the canvas and any design time needs.

Author: OK! Thanks!

[[project-version-mule-java]]
== Select Mule and Java Versions for a Project

Set Mule or Java versions for a specific implementation or integration project from *Project Properties*. You can change versions at any time, but test your projects with the versions you intend to deploy. 

The IDE uses bundled runtimes to open and edit projects immediately. The design service uses the latest Java-compatible runtime available locally.  Anypoint Code Builder bundles the latest runtime (for example, 4.11) for Java 17 and a reduced 4.8 runtime for Java 8. This guarantees that the design service can always start and serve the canvas and design-time needs without requiring runtime downloads.

> The `minMuleVersion` in your project's `mule-artifact.json` file specifies the minimum compatible Mule runtime version required to open and edit the project. The design service automatically selects the latest locally available runtime that is compatible with your project's Java version, enabling forward compatibility.

Reviewer 2: this is not very accurate. The minMuleVersion check occurs at deployment, not when opening or editing a project. Since the bundled runtime is the latest version, projects can always be opened and edited regardless of their minMuleVersion. Version compatibility is only checked during deployment.

Author: OK! Thanks!

> Because the bundled runtime is the latest version, you can open and edit any project regardless of its `minMuleVersion` property. The design service uses the latest Java-compatible runtime available locally for design-time activities. The `minMuleVersion` propertyin your project's `mule-artifact.json` file is checked only at deployment time. The `minMuleVersion` property specifies the minimum Mule runtime version required to run or deploy the application.

Reviewer 2: missing space here **propertyin**

Author: Great catch! Thanks!

Version settings are important in several scenarios:

* Updating a project to use more current versions
* Switching to a supported version, for example, for a project that you import into Anypoint Code Builder that uses an unsupported version
* Selecting versions of Mule or Java that are compatible with each other
* Specifying exact runtime versions for run and debug operations

When a Mule or Java version isn't set for a project, the IDE's canvas and notifications provide a *Set version* button that opens the *Project Properties* tab. You can also navigate directly to the *Project Properties* tab.

[NOTE]
====

> When you run or debug an application, the IDE checks if the exact runtime version specified in your project is available locally. If not, you'll see a warning dialog that allows you to:

Reviewer: ```suggestion
When you run or debug an application, the IDE checks if the exact runtime version specified in your project is available locally. If not, a warning dialog shows asking to:
```

[[project-version-mule-java]]
== Select Mule and Java Versions for a Project

Set Mule or Java versions for a specific implementation or integration project from *Project Properties*. You can change versions at any time, but test your projects with the versions you intend to deploy. 

> The IDE uses bundled Mule runtime versions to open and edit projects immediately. The design service uses the latest Java-compatible Mule runtime available locally.  Anypoint Code Builder bundles the latest Mule runtime version (for example, 4.11) for Java 17 and a reduced Mule runtime version 4.8 for Java 8. This guarantees that the design service can always start and serve the canvas and design-time needs without requiring runtime downloads.

Reviewer: ```suggestion
The IDE uses bundled Mule runtime versions to open and edit projects immediately. The design service uses the latest Java-compatible Mule runtime version available locally.  Anypoint Code Builder bundles the latest Mule runtime version (for example, 4.11) for Java 17 and a reduced Mule runtime version 4.8 for Java 8. This guarantees that the design service can always start and serve the canvas and design-time needs without requiring runtime downloads.
```

Author: OK! Thanks!

> Because the bundled Mule runtime version is the latest version, you can open and edit any project regardless of its `minMuleVersion` property. The design service uses the latest Java-compatible Mule runtime version available locally for design-time activities. The `minMuleVersion` property in your project's `mule-artifact.json` file is checked only at deployment time. The `minMuleVersion` property specifies the minimum Mule runtime version required to run or deploy the application.

Reviewer: ```suggestion
Because the bundled Mule runtime version is the latest one, you can open and edit any project regardless of its `minMuleVersion` property. The design service uses the latest Java-compatible Mule runtime version available locally for design-time activities. The `minMuleVersion` property in your project's `mule-artifact.json` file is checked only at deployment time. The `minMuleVersion` property specifies the minimum Mule runtime version required to run or deploy the application.
```

Author: OK! Thanks!

Version settings are important in several scenarios:

* Updating a project to use more current versions
* Switching to a supported version, for example, for a project that you import into Anypoint Code Builder that uses an unsupported version
* Selecting versions of Mule or Java that are compatible with each other
* Specifying exact Mule runtime versions for run and debug operations

> When you run or debug an application, the IDE checks if the exact runtime version specified in your project is available locally. If not, you see a warning dialog that allows you to:

Reviewer: ```suggestion
When you run or debug an application, the IDE checks if the exact Mule runtime version specified in your project is available locally. If not, you see a warning dialog that allows you to:
```

Author: OK! Thanks!

* Continue with the latest compatible local version
* Open *Project Properties* to select or download the required Mule runtime version
 
[NOTE]

> Changing the minimum Mule version (`minMuleVersion`) manually can have serious implications. This requires a full test of your application so it works correctly with the target Mule runtime. Newer Mule runtime versions turn on feature flags by default. See xref:mule-runtime::feature-flagging.adoc[]. This can change behavior that was fixed in later versions. Your application can rely on behavior that was later corrected or on a previous issue. Prefer changing the Mule version through *Project Properties*, and always run full regression tests after any version change.

Reviewer: ```suggestion
Changing the minimum Mule version (`minMuleVersion`) manually can have serious implications. This requires a full test of your application so it works correctly with the target Mule version. Newer Mule runtime versions turn on feature flags by default. See xref:mule-runtime::feature-flagging.adoc[]. This can change behavior that was fixed in later versions. Your application can rely on behavior that was later corrected or on a previous issue. Prefer changing the Mule version through *Project Properties*, and always run full regression tests after any version change.
```

image::int-project-properties-connectors.png["Connector versions in Project Properties tab"]
. Select connector versions. You can update Exchange connector versions for compatibility with your project's Mule runtime and Java versions.

[NOTE]
====
Connector compatibility is checked against the selected Mule runtime and Java versions. Compatibility issues appear after you apply changes in *Project Properties*.
====

[[install-additional-runtimes]]
== Install Additional Mule Runtime Versions

> Bundled runtimes enable instant project opening and design. If you must specific runtime versions for running or debugging your applications, install additional runtime versions.

Reviewer: ```suggestion
Bundled Mule runtime versions enable instant project opening and design. If you need specific Mule versions for running or debugging your applications, install these additional versions.
```
There was a word missing, not sure if I applied the correct fix. 
Global: Please check wording around "Mule runtimes", I proposed a fix according to our terminology.

Author: OK! Thanks!
